### PR TITLE
[Updater] Fix how customCourseUnitAttainments are created as courses

### PR DESCRIPTION
**Pull request by mattirasanen**
- [Backend] Fix courseldToShare exploding when providers is undefined
- [Updater] Fix how customCourseUnitAttainments are created as courses
- [FD-Compose] Replicate updater-worker
- [Course statistics] Changed toCourseRow key to use course.id instead of course.code because there could be multiple courses with same code

### CODE REVIEW

File: updater/sis-updater-worker/src/updater/updateStudents/attainments.js [hunk 1]
```diff
@@ -135,53 +135,44 @@ const updateAttainments = async (
...
-            code: parsedCourseCode,
-            coursetypecode: att.study_level_urn,
-            course_unit_type: att.course_unit_type_urn,
-          })
+        courseUnit = {
+          id: att.id,
+          name: att.name,
+          code: att.code,
+          latest_instance_date: att.attainment_date,
+          is_study_module: !(
```
Comment (valtterikantanen) [inline updater/sis-updater-worker/src/updater/updateStudents/attainments.js:149]: This should probably be
```js
is_study_module: [
  'StudyModuleAttainment',
  'CustomStudyModuleAttainment',
  'DegreeProgrammeAttainment',
  'CustomDegreeProgrammeAttainment',
].includes(att.type),
```
or better yet, add the missing module types to https://github.com/UniversityOfHelsinkiCS/oodikone/blob/548f3fc622395e143757238e41de52a0c7c3c3d4/updater/sis-updater-worker/src/updater/mapper.js#L34-L41

and export the `isModule` function. The missing custom types should also be added to `customAttainmentTypes`.

Also, there are only `ModuleAttainments` and `CustomModuleAttainments` in our db, has it been checked that `StudyModuleAttainments` and `CustomStudyModuleAttainments` also exist?

File: updater/sis-updater-worker/src/updater/updateStudents/attainments.js [hunk 2]
```diff
@@ -135,53 +135,44 @@ const updateAttainments = async (
...
           where: {
-            coursecode: course ? course.id : parsedCourseCode,
+            coursecode: att.course_unit_id,
           },
         })
 
         // If there's no courseprovider, try to create course provider
         if (!courseProvider) {
+          // parsedCourseCode but method wants groupId ? sis-importer model has no groupId :<
           const mapCourseProvider = courseProviderMapper(parsedCourseCode)
```
Comment (valtterikantanen) [inline updater/sis-updater-worker/src/updater/updateStudents/attainments.js:176]: There's now a little problem regarding how new course providers are created. `coursecode` in the course_providers table should reference the `id` column in the course table but now in custom attainments the id is actually the id from the attainment.

So maybe we should replace `parsedCourseCode` with `att.id`? This seems to fix the problem at least for us.

File: updater/sis-updater-worker/src/updater/updateStudents/attainments.js (hunk 1 above)
Comment (mattirasanen) [reply updater/sis-updater-worker/src/updater/updateStudents/attainments.js:149]: Yea checked from ori swagger and there is not `StudyModuleAttainments` and `CustomStudyModuleAttainments`

Comment (mattirasanen) [reply updater/sis-updater-worker/src/updater/updateStudents/attainments.js]: Should be ok now

File: updater/sis-updater-worker/src/updater/updateStudents/attainments.js (hunk 2 above)
Comment (mattirasanen) [reply updater/sis-updater-worker/src/updater/updateStudents/attainments.js]: Changed and seems to be working still

**Review by valtterikantanen — APPROVED** [review]

**Pull request merged**